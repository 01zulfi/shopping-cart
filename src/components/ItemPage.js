import React from "react";

const ItemPage = ({ name, price }) => (
  <section className="item-page-section">
    <h1 className="asteroid-name">{name}</h1>
    <p className="asteroid-price">Price: {price}</p>
  </section>
);

export default ItemPage;
