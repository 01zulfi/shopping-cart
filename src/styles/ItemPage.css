section.item-page-section {
  display: flex;
  flex-direction: column;
  align-items: center;
  box-shadow: 0px 15px 30px rgb(0 0 0 / 15%);
  background-color: rgba(255, 255, 255, 0.09);
  border: 1px solid #d4bff9;
  padding: 15px;
  border-radius: 10px;
  width: 95%;
  margin-top: 15px;
  margin-bottom: 25px;
}

section.item-page-section img {
  width: 75%;
}

div.content {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 15px;
  padding: 10px;
  margin-top: 0;
  text-align: center;
  gap: 10px;
}

div.content p {
  font-size: 20px;
}

div.content h1 {
  margin: 0;
  width: 75%;
  background-color: rgba(255, 255, 255, 0.11);
  text-align: center;
}

div.content .asteroid-price {
  font-size: 24px;
}

div.content a {
  font-size: 20px;
  color: white;
  font-style: italic;
}

div.break-line {
  width: 95%;
  padding: 2px;
  margin: 15px;
  background-color: #d4bff9;
}

div.content button {
  animation: button-animation 3s infinite linear alternate;
  padding: 10px;
  border: 1px solid #d4bff9;
  color: black;
  border: 0;
  font-size: 25px;
  border-radius: 5px;
}

@keyframes button-animation {
  from {
    background-color: #bc75c9;
  }
  to {
    background-color: #ab47bc;
  }
}

div.content button:hover {
  cursor: pointer;
  font-weight: bold;
}

div.content a:hover {
  color: #ab47bc;
}

@media (min-width: 768px) {
  section.item-page-section {
    width: 55%;
  }

  div.content {
    width: 85%;
  }

  section.item-page-section img {
    width: 45%;
  }
}
