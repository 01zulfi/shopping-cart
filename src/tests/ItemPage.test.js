import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ItemPage from "../components/ItemPage";

test("item page renders", () => {
  const { container } = render(<ItemPage />);
  expect(container).toMatchSnapshot();
  expect(container.firstChild).toHaveClass("item-page-section");
});

test("h1 renders", () => {
  const { container } = render(<ItemPage name="one" />);
  const name = container.querySelector("h1");
  expect(name).toHaveTextContent("one");
});
