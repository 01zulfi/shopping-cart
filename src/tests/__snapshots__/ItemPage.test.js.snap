// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`item page renders 1`] = `
<div>
  <section
    class="item-page-section"
  >
    <h1
      class="asteroid-name"
    />
    <p
      class="asteroid-price"
    >
      Price: 
    </p>
  </section>
</div>
`;
